### PR TITLE
Allow AsyncAPI mocking (WS & Kafka)

### PR DESCRIPTION
**Pull request by JulienBreux**
### Description

- WS
- Kafka

### Related issue(s)

- https://github.com/microcks/microcks-testcontainers-go/issues/8
- https://github.com/microcks/microcks-testcontainers-go/issues/9

### CODE REVIEW

File: go.mod
Comment (JulienBreux) [inline go.mod:7]: ⚠️ We're waiting for v0.31.0, blocked by release of fix https://github.com/testcontainers/testcontainers-go/pull/2551

**Comment by lbroudoux** [comment]
Related issue for Go: #8

File: ensemble/async/connection/kafka/kafka.go
Comment (lbroudoux) [inline ensemble/async/connection/kafka/kafka.go:21]: Could you explain the rationale for splitting the file into two and putting those in dedicated packages? Is it a matter of naming? Of Go conventions or best practices?

File: microcks.go
Comment (lbroudoux) [inline microcks.go:113]: Shouldn't we track this with a new issue? (just after this PR is merged)

File: microcks.go
Comment (lbroudoux) [inline microcks.go:165]: So this PR also implements #9! Nice side effect 😉

File: microcks.go
Comment (lbroudoux) [inline microcks.go:156]: We're using the `client.Secret` interface here. Shouldn't we remove the local interface type definition above?

File: ensemble/async/connection/kafka/kafka.go
Comment (JulienBreux) [reply ensemble/async/connection/kafka/kafka.go:21]: Yes in fact, this is the way in go to have "namespace" and separated concerns.

**Review by lbroudoux — CHANGES_REQUESTED** [review]
Excellent! Thanks a lot! I just have one concern regarding the definition of an interface type `Secret` in addition of the `client.Secret` that seems to be the one actually used. Can you check this?

File: microcks.go
Comment (JulienBreux) [reply microcks.go:150]: Clearly, done ✅

File: microcks.go
Comment (JulienBreux) [reply microcks.go:107]: Yes, in fact!

**Comment by JulienBreux** [comment]
@lbroudoux juste done about secret!

**Review by lbroudoux — APPROVED** [review]
LGTM!

**Pull request merged**
